packages/controls/src/controls/checkbox/checkbox.ts

```diff
@@ -0,0 +1,186 @@
+  C,
+  DataType<C>,
+  ValueType<C>,
+  ResolvedValueType<C>
+> {
+  private static readonly v1DataType = 'checkbox::v1' as const
+  private static readonly dataSignature = {
+    v1: { [ControlDataTypeKey]: this.v1DataType },
+  } as const
+
+  static readonly type = 'makeswift::controls::checkbox' as const
+
+  static get schema() {
+    const version = z.literal(1).optional()
+    const versionedData = z.object({
+      [ControlDataTypeKey]: z.literal(this.v1DataType),
+      value: z.boolean(),
+    })
+
+    const schemas = <V, D>(value: SchemaType<V>, data: SchemaType<D>) => {
+      const type = z.literal(this.type)
+
+      const config = z.object({
+        label: z.string().optional(),
+        defaultValue: value,
+      })
+
+      const definition = z.object({
+        type,
+        config,
+        version,
+      })
+
+      return {
+        type,
+        value,
+        resolvedValue: value,
+        data,
+        versionedData,
+        config,
+        version,
+        definition,
+      }
+    }
+
+    return {
+      version,
+      relaxed: schemas(
+        z.boolean().optional(),
+        z.union([z.boolean(), versionedData, z.undefined()]),
+      ),
+      strict: schemas(z.boolean(), z.union([z.boolean(), versionedData])),
+    }
+  }
+
+  static deserialize(data: DeserializedRecord): CheckboxDefinition {
+    if (data.type !== Definition.type)
+      throw new Error(
+        `Checkbox: expected '${Definition.type}', got '${data.type}'`,
+      )
+
+    const { config, version } = Definition.schema.relaxed.definition.parse(data)
+    return new CheckboxDefinition(config, version)
+  }
+
+  constructor(
+    config: C,
+    readonly version: z.infer<typeof Definition.schema.version>,
+  ) {
+    super(config)
+  }
+
+  get controlType() {
+    return Definition.type
+  }
+
+  get schema(): ReturnedSchemaType<C> {
+    return Definition.schema.relaxed
+  }
+
+  get dataSchema() {
+    return (
+      (this.config.defaultValue === undefined
+        ? Definition.schema.relaxed
+        : Definition.schema.strict) as Schema<C>
+    ).data
+  }
+
+  safeParse(data: unknown | undefined): ParseResult<DataType<C> | undefined> {
+    return safeParse(this.dataSchema, data)
+  }
+
+  fromData(data: DataType<C> | undefined): ValueType<C> | undefined {
+    const inputSchema = this.dataSchema.optional()
+    return match(data satisfies z.infer<typeof inputSchema>)
+      .with(Definition.dataSignature.v1, ({ value }) => value)
+      .otherwise((value) => value)
+  }
+
+  toData(value: ValueType<C>): DataType<C> {
+    return match(this.version)
+      .with(1, () =>
+        value === undefined
+          ? undefined
+          : {
+              ...Definition.dataSignature.v1,
+              value,
+            },
+      )
+      .with(undefined, () => value)
+      .exhaustive()
+  }
+
+  copyData(
+    data: DataType<C> | undefined,
+    _context: CopyContext,
+  ): DataType<C> | undefined {
+    return data
```

Comment:
Should we be shallowing cloning this?

Reply:
Good question, the old code did not:

<img width="1010" alt="copy" src="https://github.com/user-attachments/assets/6b3f6ba4-ace3-42cf-9bd7-e5147fbd7b07">